micro_manager/interpolation.py

```diff
@@ -0,0 +1,84 @@
+import numpy as np
+from sklearn.neighbors import NearestNeighbors
+
+
+class Interpolation:
+    def __init__(self, logger):
+
+        self._logger = logger
+
+    def get_nearest_neighbor_indices_local(
+        self,
+        all_local_coords: np.ndarray,
+        inter_point: np.ndarray,
+        k: int,
+    ) -> np.ndarray:
+        """
+        Get the indices of the k nearest neighbors of a point in a list of coordinates.
+
+        Parameters
+        ----------
+        all_local_coords : list
+            List of coordinates of all points.
+        inter_point : list | np.ndarray
+            Coordinates of the point for which the neighbors are to be found.
+        k : int
+            Number of neighbors to consider.
+
+        Returns
+        ------
+        neighbor_indices : np.ndarray
+            Indices of the k nearest neighbors in all local points.
+        """
+        assert len(all_local_coords) > 0, "No local coordinates provided."
```

Comment:
Please do a double check if this assertion makes sense here. If we never hit this case in the `micro_manager.py`, we do not need the check. Such checks typically only make sense for user-side API functions.

Reply:
Yes, this is redundant as this is ensured within the micro-manager.